lib/rack/cache/metastore.rb

```diff
@@ -38,13 +38,20 @@ def lookup(request, entity_store)
       return nil if match.nil?
 
       _, res = match
-      if body = entity_store.open(res['X-Content-Digest'])
-        restore_response(res, body)
+
+      if entity_store.nil?
+        body = ['']
       else
-        # TODO the metastore referenced an entity that doesn't exist in
-        # the entitystore. we definitely want to return nil but we should
-        # also purge the entry from the meta-store when this is detected.
+        body = entity_store.open(res['X-Content-Digest'])
+        if body.nil?
+          # TODO the metastore referenced an entity that doesn't exist in
+          # the entitystore. we definitely want to return nil but we should
```

Comment:
I'm not, but that TODO was there before my changes. I had to put that ```return nil``` there to keep the old behavior in case the entity store is not nil but no body is found for the response, otherwise some tests break.